From 2a7ab26080f9839d86788ad3c86285569a65542c Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:32 -0400
Subject: [fs] ext4: don't check quota format when there are no quota files

Message-id: <1414600485-1929-43-git-send-email-lczerner@redhat.com>
Patchwork-id: 98556
O-Subject: [RHEL 7.1 PATCH v2 42/55] ext4: don't check quota format when there are no quota files
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Jan Kara <jack@suse.cz>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit 279bf6d390933d5353ab298fcc306c391a961469

 The check whether quota format is set even though there are no
 quota files with journalled quota is pointless and it actually
 makes it impossible to turn off journalled quotas (as there's
 no way to unset journalled quota format). Just remove the check.

 CC: stable@vger.kernel.org
 Signed-off-by: Jan Kara <jack@suse.cz>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/super.c b/fs/ext4/super.c
index a0bfafe..e0759ca 100644
--- a/fs/ext4/super.c
+++ b/fs/ext4/super.c
@@ -1711,13 +1711,6 @@ static int parse_options(char *options, struct super_block *sb,
      "not specified");
    return 0;
   }
- } else {
-  if (sbi->s_jquota_fmt) {
-   ext4_msg(sb, KERN_ERR, "journaled quota format "
-     "specified with no journaling "
-     "enabled");
-   return 0;
-  }
  }
 #endif
  if (test_opt(sb, DIOREAD_NOLOCK)) {
-- 
1.7.1